frontend/packages/core/src/AppLayout/shortLinker.tsx

```diff
@@ -0,0 +1,193 @@
+
+import { Button, ClipboardButton, IconButton } from "../button";
+import { useAppContext, useShortLinkContext } from "../Contexts";
+import type { HydratedData } from "../Contexts/workflow-storage-context/types";
+import { Toast } from "../Feedback";
+import { TextField } from "../Input";
+import { client } from "../Network";
+import type { ClutchError } from "../Network/errors";
+import styled from "../styled";
+
+import { workflowByRoute } from "./utils";
+
+const Grow = styled(MuiGrow)((props: { placement: string }) => ({
+  transformOrigin: props.placement,
+}));
+
+const Popper = styled(MuiPopper)({
+  padding: "0 12px",
+  marginLeft: "12px",
+  zIndex: 1201,
+});
+
+const Paper = styled(MuiPaper)({
+  width: "400px",
+  height: "100px",
+  padding: "15px",
+  boxShadow: "0px 15px 35px rgba(53, 72, 212, 0.2)",
+  borderRadius: "8px",
+});
+
+const StyledLinkIcon = styled(IconButton)<{ $open: boolean }>(
+  {
+    color: "#ffffff",
+    marginRight: "8px",
+    padding: "12px",
+    "&:hover": {
+      background: "#2d3db4",
+    },
+    "&:active": {
+      background: "#2938a5",
+    },
+  },
+  props => ({
+    background: props.$open ? "#2d3db4" : "unset",
+  })
+);
+
+/**
+ * Component that will display a Button to generate a short link
+ * On click the component will do the following:
+ * - Read in the data from the temporary storage in the WorkflowStorageContext
+ * - Rotate it into a readable format for the API
+ * - Send an API request asking for it to be stored
+ * - If successful,
+ * -     Will switch to a readable only input field with a clipboard button to copy the short link
+ *       (this is the only time it is ever displayed)
+ * - If not successful,
+ * -     Will display a toast error message for the user
+ */
+const ShortLinker = () => {
+  const { workflows } = useAppContext();
+  const { removeWorkflowSession, retrieveWorkflowSession } = useShortLinkContext();
+  const [open, setOpen] = React.useState(false);
+  const [shortLink, setShortLink] = React.useState<string | null>(null);
+  const [validWorkflow, setValidWorkflow] = React.useState<boolean>(false);
+  const [error, setError] = React.useState<ClutchError | null>(null);
+  const anchorRef = React.useRef(null);
+  const location = useLocation();
+
+  const checkValidWorkflow = () => {
+    const workflow = workflowByRoute(workflows, location.pathname);
+
+    setValidWorkflow(workflow?.shortLink ?? false);
+  };
+
+  // will trigger on a location change, emptying out our temporary storage and rechecking the workflow
+  React.useEffect(() => {
+    if (workflows.length) {
+      checkValidWorkflow();
+    }
+
+    removeWorkflowSession();
+  }, [location]);
+
+  // Used for initial load to verify that once our workflows have loaded we are on a valid workflow
+  React.useEffect(() => {
+    if (workflows.length) {
+      checkValidWorkflow();
+    }
+  }, [workflows]);
+
+  const handleToggle = () => {
+    setOpen(!open);
+    setShortLink(null);
```

Comment:
I'm confused by this since the docstring says that if there is an existing shortlink there shouldn't be an option to generate a new one. However, if the user opens the popper, generates a shortlink, and then closes it the shortlink will be set to null and a new one will need to be generated.

Reply:
Which docstring are you referring to? The clearing was meant so that a user could build off of a shortlink. Generate 1, modify modify, generate 2, modify modify, etc etc. The docstring about `only time it is ever displayed` is more in line with what this does in that after you create it it will display it so you can copy and then it's gone (similar to how generating a private key works in most services).

Unless Im misunderstanding.

Reply:
ah ok I see the confusion. The docstring implies that this link is going to be persisted.
This probably opens up another question which is, is there a reason to unset the shortlink if nothing in the state has changed? We don't really track that at the moment but if the state doesn't change the link should remain the same. Not something we have to worry about in this PR but it is something we should probably do as a follow up

Reply:
Yeah that's probably a good idea, shouldn't be too hard to adjust either